crates/hir-def/src/attr.rs

```diff
@@ -202,6 +202,12 @@ impl Attrs {
                 .rev()
                 .zip(["core", "prelude", "v1", "test"].iter().rev())
                 .all(|it| it.0.as_str() == Some(it.1))
+        }) || self.iter().any(|it| {
+            it.path()
+                .segments()
+                .iter()
+                .zip(["tokio", "test"].iter())
+                .all(|it| it.0.as_str() == Some(it.1))
```

Comment:
That's the wrong way to fix this, we need to go up the expansion hierarchy to find the necessary test attribute when categorizing the references

Reply:
I thought that's too slow as mentioned here: https://github.com/rust-lang/rust-analyzer/pull/16441#discussion_r1471087307

Although I would prefer to have a full filtered result which takes a little bit longer

Reply:
Right, that was about general traversal which we do, we are missing traversing up the macro expansions still though. Seems that was overlooked in the PR. That won't make much of a difference I guess given most finds shouldn't be in macro calls anyways.

Reply:
I thought it's related to `tokio::test`, but it's not:
https://github.com/rust-lang/rust-analyzer/pull/16768#discussion_r1514463398